src/Form/LegacyConsumer/resources/js/conditinal-fields.js

```diff
@@ -0,0 +1,193 @@
+			const visibilityCondition = visibilityConditions[0]; // Currently we support only one visibility condition.
+			const {field} = visibilityCondition;
+
+			fields[field] = {
+				...fields[field],
+				[inputField.name]: visibilityConditions
+			}
+		})
+
+		return fields;
+	}
+
+	/**
+	 * @unreleased
+	 *
+	 * @param operator
+	 * @param firstData
+	 * @param secondData
+	 *
+	 * @return boolean
+	 */
+	function compareWithOperator( operator, firstData, secondData ){
+		return {
+			'=': firstData === secondData,
+			'!=': firstData != secondData,
+			'>': firstData > secondData,
+			'>=': firstData >= secondData,
+			'<': firstData < secondData,
+			'<=': firstData <=secondData
+		}[operator]
+	}
+
+	/**
+	 * Handle fields visibility.
+	 * @unreleased
+	 */
+	function handleVisibility(donationForm, visibilityConditionsForWatchedField) {
+		for (const [inputFieldName, visibilityConditions] of Object.entries(visibilityConditionsForWatchedField)) {
+			const inputField = donationForm.querySelector(`[name="${inputFieldName}"]`);
+			const fieldWrapper = inputField.closest('.form-row');
+			const visibilityCondition = visibilityConditions[0]; // Currently we support only one visibility condition.
+			let visible = false;
+			const {field, operator, value} = visibilityCondition;
+
+			const inputs = donationForm.querySelectorAll(`[name="${field}"]`);
+			let hasFieldController = !!inputs.length;
+
+			if (hasFieldController) {
+				inputs.forEach((input) => {
+					const fieldType = input.getAttribute('type');
+					const comparisonResult = compareWithOperator(operator, input.value, value);
+
+					if (fieldType && (fieldType === 'radio' || fieldType === 'checkbox')) {
+						if (input.checked && comparisonResult) {
+							visible = true;
+						}
+					} else if (comparisonResult) {
+						visible = true;
+					}
+				});
+
+				// Show or Hide field wrapper.
+				visible ?
+					fieldWrapper.classList.remove('give-hidden') :
+					fieldWrapper.classList.add('give-hidden');
+			}
+		}
+	}
+
+	/**
+	 * Setup state for condition visibility settings.
+	 * state contains list of watched elements per donation form.
+	 *
+	 * @unreleased
+	 */
+	function addVisibilityConditionsToStateForDonationForm(donationForm) {
+		const uniqueDonationFormId = donationForm.getAttribute('data-id');
+		const watchedFields = getWatchedElementNames(donationForm);
+
+		// Add donation form to state only if visibility conditions exiting for at least form field.
+		if (uniqueDonationFormId && Object.keys(watchedFields).length) {
+			state[uniqueDonationFormId] = watchedFields;
+		}
+	}
+
+	/**
+	 * Setup state for condition visibility settings.
+	 * state contains list of watched elements per donation form.
+	 *
+	 * @unreleased
+	 */
+	async function addVisibilityConditionsToStateForAllDonationForm() {
+		await document.querySelectorAll('form.give-form').forEach(addVisibilityConditionsToStateForDonationForm);
```

Comment:
@ravinderk querySelectorAll does not return a promise so you don't need to use `async` `await`. Check [docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll#return_value) for more info.

Reply:
fixed: https://github.com/impress-org/givewp/compare/72ed8c4845c6...cce2a1f53929